### PR TITLE
Feature/icon enums

### PR DESCRIPTION
**Pull request by tayambamwanza**
- Moved ngx-bootstrap-icons into it's own folder in preparation of adding a new component
- Add optional enum support for <i-bs>

### CODE REVIEW

**Comment by avmaisak** [comment]
@tayambamwanza project have tool script **generate-icons.js** which generates icons.
this enum must be generated too, using this script

**Comment by tayambamwanza** [comment]
@avmaisak I wrote a custom node script should I migrate the code into that file then?

**Review by avmaisak — COMMENTED** [review]
> @avmaisak I wrote a custom node script should I migrate the code into that file then?

there are script in path **tools/generate-icons.js**  it runs when executing **npm run gen**.

for consistence enum content must be created from this script, using command **npm run gen** too.

**Comment by tayambamwanza** [comment]
> > @avmaisak I wrote a custom node script should I migrate the code into that file then?
> 
> there are script in path **tools/generate-icons.js** it runs when executing **npm run gen**.
> 
> for consistence enum content must be created from this script, using command **npm run gen** too.

I will see what I can do, will try something now.

File: projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts [hunk 1]
```diff
@@ -0,0 +1,1153 @@
+export enum IconNamesEnum {
```
[the remark this answers is not in the document]
Comment (tayambamwanza) [reply projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts:1]: I will see what I can do, will try something now.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+import { IconNamesEnum } from 'projects/ngx-bootstrap-icons-lib/src/public-api';
 import { icons } from './icons';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
 })
 export class AppComponent implements OnInit {
+
+  public IconNames = IconNamesEnum;
```
[the remark this answers is not in the document]
Comment (tayambamwanza) [reply src/app/app.component.ts:13]: Left over from when I was testing, I can remove it.

File: projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tayambamwanza) [reply projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts:1]: When you say the root do you mean in the src folder or lib folder?

[the remark this answers is not in the document]
Comment (avmaisak) [reply projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts:1]: image

[the remark this answers is not in the document]
Comment (avmaisak) [reply projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts:1]: lib
image

File: src/app/app.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (avmaisak) [reply src/app/app.component.ts:13]: please, remove it

File: projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tayambamwanza) [reply projects/ngx-bootstrap-icons-lib/src/lib/utils/icon-names.enum.ts:1]: Thank you

File: src/app/app.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (avmaisak) [reply src/app/app.component.ts:13]: the demo app needs to be deep refactored

File: .gitignore [hunk 3]
```diff
@@ -40,6 +40,7 @@ npm-debug.log
 yarn-error.log
 testem.log
 /typings
+/icons
```
Comment (avmaisak) [inline .gitignore:43]: no need this line

**Review by avmaisak — APPROVED** [review]

**Pull request merged**